Source/WebKit/UIProcess/ios/forms/WKFileUploadPanel.mm

```diff
@@ -778,42 +811,60 @@ - (void)_adjustMediaCaptureType
     if ([UIImagePickerController isCameraDeviceAvailable:UIImagePickerControllerCameraDeviceFront] || [UIImagePickerController isCameraDeviceAvailable:UIImagePickerControllerCameraDeviceRear]) {
         if (![UIImagePickerController isCameraDeviceAvailable:UIImagePickerControllerCameraDeviceFront])
             _mediaCaptureType = WebCore::MediaCaptureTypeEnvironment;
-        
+
         if (![UIImagePickerController isCameraDeviceAvailable:UIImagePickerControllerCameraDeviceRear])
             _mediaCaptureType = WebCore::MediaCaptureTypeUser;
-        
+
         return;
     }
-    
+
     _mediaCaptureType = WebCore::MediaCaptureTypeNone;
 }
 
 - (BOOL)_shouldMediaCaptureOpenMediaDevice
 {
     if (_mediaCaptureType == WebCore::MediaCaptureTypeNone || ![UIImagePickerController isSourceTypeAvailable:UIImagePickerControllerSourceTypeCamera])
         return NO;
-    
+
     return YES;
 }
 
-- (void)_showPhotoPickerWithSourceType:(UIImagePickerControllerSourceType)sourceType
+- (void)_showCamera
 {
-    ASSERT([UIImagePickerController isSourceTypeAvailable:sourceType]);
-
-    _imagePicker = adoptNS([[UIImagePickerController alloc] init]);
-    [_imagePicker setSourceType:sourceType];
-    [_imagePicker setMediaTypes:[self _mediaTypesForPickerSourceType:sourceType]];
-    [_imagePicker setDelegate:self];
-    [_imagePicker presentationController].delegate = self;
-    [_imagePicker setAllowsEditing:NO];
-    [_imagePicker _setAllowsMultipleSelection:_allowMultipleFiles];
-    [_imagePicker _setRequiresPickingConfirmation:YES];
-    [_imagePicker _setShowsFileSizePicker:YES];
+    ASSERT([UIImagePickerController isSourceTypeAvailable:UIImagePickerControllerSourceTypeCamera]);
+
+    _cameraPicker = adoptNS([[UIImagePickerController alloc] init]);
+    [_cameraPicker setSourceType:UIImagePickerControllerSourceTypeCamera];
+    [_cameraPicker setMediaTypes:[self _mediaTypesForPickerSourceType:UIImagePickerControllerSourceTypeCamera]];
+    [_cameraPicker setDelegate:self];
+    [_cameraPicker presentationController].delegate = self;
+    [_cameraPicker setAllowsEditing:NO];
+    [_cameraPicker _setAllowsMultipleSelection:_allowMultipleFiles];
+    [_cameraPicker _setRequiresPickingConfirmation:YES];
+    [_cameraPicker _setShowsFileSizePicker:YES];
 
     if (_mediaCaptureType != WebCore::MediaCaptureTypeNone)
-        [_imagePicker setCameraDevice:cameraDeviceForMediaCaptureType(_mediaCaptureType)];
+        [_cameraPicker setCameraDevice:cameraDeviceForMediaCaptureType(_mediaCaptureType)];
+
+    [self _presentFullscreenViewController:_cameraPicker.get() animated:YES];
+}
+
+- (void)_showPhotoPicker
+{
+#if HAVE(PHOTOS_UI)
+    auto configuration = adoptNS([allocPHPickerConfigurationInstance() init]);
```

Comment:
We should set `PHPickerConfigurationAssetRepresentationModeCompatible` on the configuration, for compatibility. The `Automatic` (default) value is subject to change.

If possible, we should also remove all the logic I added in https://bugs.webkit.org/show_bug.cgi?id=230639, if `PHPicker` does it for free.

Reply:
Isn't all that logic still needed for the camera case though?

Reply:
That's something to test – the logic was originally added for multiple selection mode (see the Changelog).